src/Surfnet/StepupSelfService/SelfServiceBundle/Controller/RegistrationController.php

```diff
@@ -31,11 +31,17 @@ class RegistrationController extends Controller
      */
     public function displaySecondFactorTypesAction()
     {
-        $enabledSecondFactors = $this->getParameter('ss.enabled_second_factors');
+        $institutionConfigurationOptions = $this->get('self_service.service.institution_configuration_options')
+            ->getInstitutionConfigurationOptionsFor($this->getIdentity()->institution);
+
+        $allowedSecondFactors = array_intersect(
```

Comment:
now I get why it is an String[] rather than SecondFactorType[] 😉 

Nice and proper use of array_intersect! Wish there was a more elegant solution but for now this'll suffice.